### PR TITLE
CLOUD-2788 / CLOUD-2546 - support for upstream KUBE_PING and DNS_PING

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.jboss.org/browse/CLOUD-2788
https://issues.jboss.org/browse/CLOUD-2546

This fixes the documentation generation to include the clustering parts.

Signed-off-by: Ken Wills kwills@redhat.com

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by luck3y** [comment]
@bstansberry This one is just missing docs.

File: gen_template_docs.py [hunk 1]
```diff
@@ -148,9 +148,7 @@ def createTemplate(data, path):
                     if len(secretName) > 0:
  	                  tdata['objects'][0]['secrets'] = [{ "secretName": secretName[0], "secretFile": secretName[0] + ".json" }]
 		    
-        # currently the clustering section applies only to EAP templates
-        if re.match('^eap', path):
```
Comment (bstansberry) [inline gen_template_docs.py:152]: Is this not needed because now the entire file is scoped to the EAP image?

Comment (luck3y) [reply gen_template_docs.py:152]: Correct. The script originated in the application-templates repo, so was filtering based on image names etc.

**Comment by bstansberry** [comment]
Can you add CLOUD-2546 to the PR title?

I ask because the PR title becomes part of the merge commit message, and it's nice to find the merge commits (and from those the PR #) by grepping the log. I do this a lot:

```
taozi:jboss-eap-7-openshift-image bstansberry$ git log | grep -B 5 CLOUD-2729
commit a83a42fed52550792fafaa8e514e42951547769f
Merge: fa126b4 3c1c499
Author: Brian Stansberry <brian.stansberry@redhat.com>
Date:   Tue Sep 18 15:38:56 2018 -0500

    Merge pull request #136 from bstansberry/CLOUD-2729
--
    
     [CLOUD-2729] Integrate the jboss.eap.config.mp-config module
--

commit 3c1c499e0eb04b41eebd219e25148dbd6ab37f30
Author: Brian Stansberry <brian.stansberry@redhat.com>
Date:   Fri Aug 31 17:36:40 2018 -0500

    [CLOUD-2729] Integrate the jboss.eap.config.mp-config module
```

^^^ isn't a great example because my branch was named CLOUD-2729, so the grep would find the merge commit anyway, but, anyway...

File: docs/templates/eap-cd-amq-persistent-s2i.adoc [hunk 2]
```diff
@@ -292,4 +292,76 @@ more information.
 
 
 
+[[clustering]]
+==== Clustering
+
+Clustering in OpenShift EAP is achieved through one of two discovery mechanisms:
+Kubernetes or DNS. This is done by configuring the JGroups protocol stack in
+standalone-openshift.xml with either the `<openshift.KUBE_PING/>` or `<openshift.DNS_PING/>`
+elements. The templates are configured to use `DNS_PING`, however `KUBE_PING`is
```
Comment (bstansberry) [inline docs/templates/eap-cd-amq-persistent-s2i.adoc]: This doesn't discuss upstream KUBE_PING and DNS_PING.

Should this wait until the jboss-eap-modules change is in?  Or is this not really about CLOUD-2788 / 2546?

Comment (luck3y) [reply docs/templates/eap-cd-amq-persistent-s2i.adoc]: That's a good question. Do we want to specifically document the 4 ping implementations available now? The key bit that the docs talk about are the service account permissions for KUBE_PING (the same permissions apply to both implementations.)

Answering my own question, I think we should mention them in the docs, especially for CD, I'll update.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**